vignettes/speedup_options.Rmd

```diff
@@ -0,0 +1,710 @@
+)
+
+# 7-day forecast window
+horizon <- 7
+
+# Rt default prior
+rt_prior_default <- list(mean = 2, sd = 0.1)
+```
+
+Now, let's generate the `estimates` object from `epinow()`.
+
+``` r
+estimates <- epinow(
+  data = example_confirmed[1:100],
+  generation_time = generation_time_opts(example_generation_time),
+  delays = delay_opts(example_incubation_period + example_reporting_delay),
+  rt = rt_opts(prior = rt_prior_default, rw = 14),
+  gp = NULL,
+  obs = obs,
+  horizon = 0, # no forecasting
+  output = "fit"
+)
+#> DEBUG [2024-08-07 14:41:13] epinow: Running in exact mode for 2000 samples (across 4 chains each with a warm up of 250 iterations each) and 114 time steps of which 0 are a forecast
+```
+
+For the `R` data, we will set up an arbitrary trajectory and add some Gaussian noise.
+
+``` r
+# Arbitrary reproduction number trajectory
+R <- c(
+  rep(2, 40), rep(0.5, 10), rep(1, 10), 1 + 0.04 * 1:20, rep(1.4, 5),
+  1.4 - 0.02 * 1:20, rep(1.4, 10), rep(0.8, 5), 0.8 + 0.02 * 1:20
+)
+# Add Gaussian noise
+R_noisy <- R * rnorm(length(R), 1, 0.05)
+```
+
+Now, we are ready to simulate the true infections and $R_t$ data by sampling from $10$ posterior samples.
+
+``` r
+# Forecast infections and the trajectory of Rt
+forecast <- forecast_infections(
+  estimates$estimates,
+  R = R_noisy,
+  samples = 10
+)
+#> DEBUG [2024-08-07 14:41:54] simulate_infections: Running in exact mode for 1 samples (across 1 chains each with a warm up of 1 iterations each) and 154 time steps of which 40 are a forecast
+```
+
+We will now extract the required data:
+- `R_true`: the median of the simulated $R_t$ values,
+- `infections_true`: the infections by date of infection, and
+- `reported_cases_true`: the reported cases by date of report.
+
+``` r
+R_true <- forecast$summarised[variable == "R"]$median
+
+# Get the posterior samples from which to extract the simulated infections and reported cases
+posterior_sample <- forecast$samples[sample == 1]
+
+# Extract the simulated infections
+infections_true <- posterior_sample[variable == "infections"]$value
+
+# Extract the simulated reported cases and rename the "value" column to "confirm" (to match EpiNow2 requirements)
+reported_cases_true <- posterior_sample[
+  variable == "reported_cases", .(date, confirm = value)
+]
+```
+
+Now, to the main part of this vignette: we will define and run the different model options and evaluate their runtimes and estimation and forecasting performance.
+
+## Model options
+
+Below we describe each model.
+
+
+Table: Model options
+
+|model               |description                                                                                                            |
+|:-------------------|:----------------------------------------------------------------------------------------------------------------------|
+|default_mcmc_rstan  |Default model; Gaussian process on new infection and with a non-stationary prior on $R_t$                              |
```

Comment:
The default only has a GP on the residuals of Rt (i.e this is what makes its Rt estimates non-stationary). Suggest a reword here as it sounds like its on new infections.